### ISSUE TITLE
Consider renaming registerShader to registerMaterial

### ISSUE THREAD

**Issue by ngokevin** [aframevr/aframe#1467]
**Description:**
- `registerShader` API is actually registering materials. Even for custom shaders, it is using `ShaderMaterial`.
- Confusing that `registerShader` maps to providing options for the material component. It's pretty tough in the docs to explain this.

Related: https://github.com/aframevr/aframe/issues/1457

**Comment by wizgrav** [comment]
I was kind of confused when I first saw it until I realized that it wraps the whole material definition so registerMaterial does make more sense, especially for devs who are already acquainted with three.js. The disambiguation with the schema's "type" property is not so nice though. Maybe a synonym like "kind" or something could be used to define the material and geometry components variations.

**Comment by ngokevin** [comment]
How might `type` be ambiguous? It maps to `THREE.Material.type`.

**Comment by ngokevin** [comment]
@dmarcos Thoughts? I think this will be less confusing.

**Comment by wizgrav** [comment]
it would be ambiguous when you define the schema parser, there's already a type there

**Comment by ngokevin** [comment]
`type` isn't actually reserved:

This is single-prop:

```js
schema: {
  type: 'string'
}
```

This is multi-prop:

```js
schema: {
  type: {default: 'string'}
}
```

**Comment by wizgrav** [comment]
hmm then that's a non issue I guess

**Comment by ngokevin** [comment]
@dmarcos Any objections?

**Comment by dmarcos** [comment]
I would keep registerShader. Material use shaders and it's what you select on the shader property. It's also consistent with the registerPrimitive API for geometries.

**Comment by dmarcos** [comment]
Shaders is an advanced feature. We have to document it but it's not necessary to dumb it down.

**Comment by dmarcos** [comment]
Shader is a wide spread computer graphics concept. ShaderMaterial is specific to three.js

**Comment by ngokevin** [comment]
Yeah, materials use shaders technically, but API is an abstraction on top of three.js materials and an extension to the material component. registerMaterial registering materials and is way more clear than registerSbader registering materials.

geometry is registerGeometry, not registerPrimitive. 

For documentation, I'm not referring to documenting custom shaders, I'm referring to documenting our materials that we refer to as Shaders. Readers would not expect that A-Frame materials map to A-Frame shaders.

Just doesn't make sense to break a perfect mapping.
